### PR TITLE
unit test cases for combobox

### PR DESCRIPTION
**Pull request by Neha-Singla**
https://github.com/PublicisSapient/enable-a11y/issues/120

testcases for ComboBox linked to the page 
https://www.useragentman.com/enable/combobox.php

### CODE REVIEW

File: js/test/combobox.test.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    expect(results.ariaOwns).toBe(expectedAttributes.ariaOwns);
+    expect(results.ariaAutocomplete).toBe(expectedAttributes.ariaAutocomplete);
+}
+
+describe('Combobox Test', () => {
+    beforeAll(async () => {
+        await page.goto(`${config.BASE_URL}/combobox.php`);
+    });
+
+    it('HTML5 Native combobox is not completely keyboard accessible', async () => {
```
Comment (zoltan-dulac) [inline js/test/combobox.test.js:32]: I thought I mentioned this, but it probably forgot: you don't need to do unit testing on the native  HTML5 versions of components.  Only the custom ARIA versions.

I'll mention this in the next meeting.  This should be true for all unit testing.

File: js/test/combobox.test.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        );
+
+        await page.waitForSelector('#languages');
+        const datalistCount = Array.from(await page.$$('#languages')).length;
+        expect(valueAfterEnter).toBe('');
+        expect(datalistCount).toBe(1);
+    });
+
+    it('ARIA Combobox  is completely keyboard accessible', async () => {
+        await page.waitForSelector('#aria-fruit');
```
Comment (zoltan-dulac) [inline js/test/combobox.test.js:70]: I'd like you to test all ARIA comboboxes at the same time (there are two).

File: js/test/combobox.test.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        // Type some characters in the combobox
+        await page.type('#aria-fruit', 'app', { delay: 100 });
+        // Testing Attributes of the Active Combobox
+        const expectedAttributes = {
+            ariaDescribedBy: 'aria-fruit__desc',
+            ariaExpanded: 'true',
+            role: 'combobox',
+            ariaOwns: 'aria-fruit__list',
+            ariaAutocomplete: 'list',
+        };
```
Comment (zoltan-dulac) [inline js/test/combobox.test.js:81]: Testing the specific values of `role` and `ariaAutocomplete` is good.  

For the others, I want to make sure:

`ariaDescribedby` is set and that the id exists on the page.
`ariaExpanded` is set to an appropriate value 
`ariaOwns` is set and that the id exists on the page and that its a proper list.

File: js/test/combobox.test.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                const arr = el.value.split(',').map((item) => item.trim());
+                return { value: arr[0], category: arr[1] }; // containing value and category
+            },
+        );
+        expect(selected.value).toBe('Canada');
+        expect(selected.category).toBe('Other States group');
+    });
+
+    afterAll(async () => {});
+});
```
Comment (zoltan-dulac) [inline js/test/combobox.test.js:167]: Other things you should test:

1. Does the "Clear" button work.
2. Does the "Escape" key work
3. If I type in a string, does the number of items get populated correctly?  How about the screen reader aria-live announcement?
4. Can we check the markup for the combobox with categories?

**Review by zoltan-dulac — CHANGES_REQUESTED** [review]

File: js/test/combobox.test.js (hunk 1 above)
Comment (Neha-Singla) [reply js/test/combobox.test.js:32]: done

File: js/test/combobox.test.js (hunk 2 above)
Comment (Neha-Singla) [reply js/test/combobox.test.js:70]: i am testing the all three combo boxes. need to check on this

File: js/test/combobox.test.js (hunk 3 above)
Comment (Neha-Singla) [reply js/test/combobox.test.js:81]: ariaDescribedby is set and that the id exists on the page.--done

File: js/test/combobox.test.js (hunk 2 above)
Comment (Neha-Singla) [reply js/test/combobox.test.js:62]: done

File: js/test/combobox.test.js (hunk 4 above)
Comment (Neha-Singla) [reply js/test/combobox.test.js:233]: done

File: js/test/combobox.test.js [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+            return {
+                ariaExpanded,
+                selectedValue,
+                focusedElementType,
+            };
+        });
+        expect(domInfo.ariaExpanded).toBe('false');
+        expect(domInfo.selectedValue).toBe('Apple');
+        expect(domInfo.focusedElementType).toBe('reset');
```
Comment (zoltan-dulac) [inline js/test/combobox.test.js:105]: This is a good test, but I would also like another test where you type in a half completed word (e.g. "Ap") and choose one of the autocomplete suggestions (e.g. "Apple").  

Also you should test to see how many autocomplete suggestions it does suggest and make sure it is accurate.

Comment (Neha-Singla) [reply js/test/combobox.test.js:143]: done

**Review by zoltan-dulac — APPROVED** [review]

**Pull request merged**